mlir-sycl/lib/Conversion/SYCLToSPIRV/SYCLToSPIRV.cpp

```diff
@@ -180,6 +180,46 @@ template <typename OpTy> class GridOpPatternIndex : public GridOpPattern<OpTy> {
   }
 };
 
+template <unsigned Dimensions>
+static std::enable_if_t<(1 <= Dimensions && Dimensions < 4), int64_t>
+mirrorIndex(int64_t index);
+
+template <> int64_t mirrorIndex<1>(int64_t index) {
+  assert(index == 0 && "Invalid index");
```

Comment:
As we're taking `int` here, should we also be testing for positive values?

Reply:
Good point. Changed!